TST: test_angle_arrays failed with ValueError: setting an array element with a sequence

Looks like something in numpy-dev "nightly wheel" is incompatible with this Angle test. Example log: https://github.com/astropy/astropy/runs/7454435493?check_suite_focus=true

```
Numpy: 1.24.0.dev0+569.gf1e50253a
...
______________________________ test_angle_arrays _______________________________
    def test_angle_arrays():
        """
        Test arrays values with Angle objects.
        """
        # Tests incomplete
        a1 = Angle([0, 45, 90, 180, 270, 360, 720.], unit=u.degree)
        npt.assert_almost_equal([0., 45., 90., 180., 270., 360., 720.], a1.value)
        a2 = Angle(np.array([-90, -45, 0, 45, 90, 180, 270, 360]), unit=u.degree)
        npt.assert_almost_equal([-90, -45, 0, 45, 90, 180, 270, 360],
                                a2.value)
        a3 = Angle(["12 degrees", "3 hours", "5 deg", "4rad"])
        npt.assert_almost_equal([12., 45., 5., 229.18311805],
                                a3.value)
        assert a3.unit == u.degree
        a4 = Angle(["12 degrees", "3 hours", "5 deg", "4rad"], u.radian)
        npt.assert_almost_equal(a4.degree, a3.value)
        assert a4.unit == u.radian
        a5 = Angle([0, 45, 90, 180, 270, 360], unit=u.degree)
        a6 = a5.sum()
        npt.assert_almost_equal(a6.value, 945.0)
        assert a6.unit is u.degree
        with ExitStack() as stack:
            stack.enter_context(pytest.raises(TypeError))
            # Arrays where the elements are Angle objects are not supported -- it's
            # really tricky to do correctly, if at all, due to the possibility of
            # nesting.
            if not NUMPY_LT_1_19:
                stack.enter_context(
                    pytest.warns(DeprecationWarning,
                                 match='automatic object dtype is deprecated'))
>           a7 = Angle([a1, a2, a3], unit=u.degree)
../../.tox/py38-test-devdeps/lib/python3.8/site-packages/astropy/coordinates/tests/test_arrays.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py38-test-devdeps/lib/python3.8/site-packages/astropy/coordinates/angles.py:141: in __new__
    return super().__new__(cls, angle, unit, dtype=dtype, copy=copy,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cls = <class 'astropy.coordinates.angles.Angle'>
value = [array([  0.,  45.,  90., 180., 270., 360., 720.]), array([-90., -45.,   0.,  45.,  90., 180., 270., 360.]), array([ 12.        ,  45.        ,   5.        , 229.18311805])]
unit = Unit("deg"), dtype = None, copy = True, order = None, subok = False
ndmin = 0
    def __new__(cls, value, unit=None, dtype=np.inexact, copy=True, order=None,
                subok=False, ndmin=0):
        if unit is not None:
            # convert unit first, to avoid multiple string->unit conversions
            unit = Unit(unit)
        # inexact -> upcast to float dtype
        float_default = dtype is np.inexact
        if float_default:
            dtype = None
        # optimize speed for Quantity with no dtype given, copy=False
        if isinstance(value, Quantity):
            if unit is not None and unit is not value.unit:
                value = value.to(unit)
                # the above already makes a copy (with float dtype)
                copy = False
            if type(value) is not cls and not (subok and
                                               isinstance(value, cls)):
                value = value.view(cls)
            if float_default and value.dtype.kind in 'iu':
                dtype = float
            return np.array(value, dtype=dtype, copy=copy, order=order,
                            subok=True, ndmin=ndmin)
        # Maybe str, or list/tuple of Quantity? If so, this may set value_unit.
        # To ensure array remains fast, we short-circuit it.
        value_unit = None
        if not isinstance(value, np.ndarray):
            if isinstance(value, str):
                # The first part of the regex string matches any integer/float;
                # the second parts adds possible trailing .+-, which will break
                # the float function below and ensure things like 1.2.3deg
                # will not work.
                pattern = (r'\s*[+-]?'
                           r'((\d+\.?\d*)|(\.\d+)|([nN][aA][nN])|'
                           r'([iI][nN][fF]([iI][nN][iI][tT][yY]){0,1}))'
                           r'([eE][+-]?\d+)?'
                           r'[.+-]?')
                v = re.match(pattern, value)
                unit_string = None
                try:
                    value = float(v.group())
                except Exception:
                    raise TypeError('Cannot parse "{}" as a {}. It does not '
                                    'start with a number.'
                                    .format(value, cls.__name__))
                unit_string = v.string[v.end():].strip()
                if unit_string:
                    value_unit = Unit(unit_string)
                    if unit is None:
                        unit = value_unit  # signal no conversion needed below.
            elif isiterable(value) and len(value) > 0:
                # Iterables like lists and tuples.
                if all(isinstance(v, Quantity) for v in value):
                    # If a list/tuple containing only quantities, convert all
                    # to the same unit.
                    if unit is None:
                        unit = value[0].unit
                    value = [q.to_value(unit) for q in value]
                    value_unit = unit  # signal below that conversion has been done
                elif (dtype is None and not hasattr(value, 'dtype')
                      and isinstance(unit, StructuredUnit)):
                    # Special case for list/tuple of values and a structured unit:
                    # ``np.array(value, dtype=None)`` would treat tuples as lower
                    # levels of the array, rather than as elements of a structured
                    # array, so we use the structure of the unit to help infer the
                    # structured dtype of the value.
                    dtype = unit._recursively_get_dtype(value)
        if value_unit is None:
            # If the value has a `unit` attribute and if not None
            # (for Columns with uninitialized unit), treat it like a quantity.
            value_unit = getattr(value, 'unit', None)
            if value_unit is None:
                # Default to dimensionless for no (initialized) unit attribute.
                if unit is None:
                    unit = cls._default_unit
                value_unit = unit  # signal below that no conversion is needed
            else:
                try:
                    value_unit = Unit(value_unit)
                except Exception as exc:
                    raise TypeError("The unit attribute {!r} of the input could "
                                    "not be parsed as an astropy Unit, raising "
                                    "the following exception:\n{}"
                                    .format(value.unit, exc))
                if unit is None:
                    unit = value_unit
                elif unit is not value_unit:
                    copy = False  # copy will be made in conversion at end
>       value = np.array(value, dtype=dtype, copy=copy, order=order,
                         subok=True, ndmin=ndmin)
E       ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (3,) + inhomogeneous part.
../../.tox/py38-test-devdeps/lib/python3.8/site-packages/astropy/units/quantity.py:508: ValueError
```

@mhvk , any advise? 🙏

--- a/astropy/coordinates/tests/test_arrays.py
+++ b/astropy/coordinates/tests/test_arrays.py
@@ -8,12 +8,11 @@ from numpy import testing as npt
 from astropy import units as u
 from astropy.time import Time
 from astropy.tests.helper import assert_quantity_allclose as assert_allclose
-from astropy.utils.compat import NUMPY_LT_1_19
+from astropy.utils.compat import NUMPY_LT_1_19, NUMPY_LT_1_24
 from astropy.utils.exceptions import AstropyDeprecationWarning
 
 from astropy.coordinates import (Angle, ICRS, FK4, FK5, Galactic, SkyCoord,
                                  CartesianRepresentation)
-from astropy.coordinates.angle_formats import dms_to_degrees, hms_to_hours
 
 
 def test_angle_arrays():
@@ -43,14 +42,17 @@ def test_angle_arrays():
     assert a6.unit is u.degree
 
     with ExitStack() as stack:
-        stack.enter_context(pytest.raises(TypeError))
-        # Arrays where the elements are Angle objects are not supported -- it's
-        # really tricky to do correctly, if at all, due to the possibility of
-        # nesting.
-        if not NUMPY_LT_1_19:
-            stack.enter_context(
-                pytest.warns(DeprecationWarning,
-                             match='automatic object dtype is deprecated'))
+        if NUMPY_LT_1_24:
+            stack.enter_context(pytest.raises(TypeError))
+            # Arrays where the elements are Angle objects are not supported -- it's
+            # really tricky to do correctly, if at all, due to the possibility of
+            # nesting.
+            if not NUMPY_LT_1_19:
+                stack.enter_context(
+                    pytest.warns(DeprecationWarning,
+                                 match='automatic object dtype is deprecated'))
+        else:
+            stack.enter_context(pytest.raises(ValueError))
 
         a7 = Angle([a1, a2, a3], unit=u.degree)
 

--- a/astropy/utils/compat/numpycompat.py
+++ b/astropy/utils/compat/numpycompat.py
@@ -9,7 +9,7 @@ from astropy.utils import minversion
 
 __all__ = ['NUMPY_LT_1_19', 'NUMPY_LT_1_19_1', 'NUMPY_LT_1_20',
            'NUMPY_LT_1_21_1', 'NUMPY_LT_1_22', 'NUMPY_LT_1_22_1',
-           'NUMPY_LT_1_23']
+           'NUMPY_LT_1_23', 'NUMPY_LT_1_24']
 
 # TODO: It might also be nice to have aliases to these named for specific
 # features/bugs we're checking for (ex:
@@ -20,4 +20,5 @@ NUMPY_LT_1_20 = not minversion(np, '1.20')
 NUMPY_LT_1_21_1 = not minversion(np, '1.21.1')
 NUMPY_LT_1_22 = not minversion(np, '1.22')
 NUMPY_LT_1_22_1 = not minversion(np, '1.22.1')
-NUMPY_LT_1_23 = not minversion(np, '1.23dev0')
+NUMPY_LT_1_23 = not minversion(np, '1.23')
+NUMPY_LT_1_24 = not minversion(np, '1.24dev0')

